### PR TITLE
Add print methods for rest of custom classes

### PR DESCRIPTION
**Pull request by nmdefries**
Closes https://github.com/cmu-delphi/epidatr/issues/51

### CODE REVIEW

File: R/epidatacall.R [hunk 1]
```diff
@@ -198,6 +198,14 @@ fetch_args_list <- function(
   )
 }
 
+#' @export
+print.fetch_args <- function(x, ...) {
+  stopifnot(inherits(x, "fetch_args"))
+  cli::cli_h1("<fetch_args> object:")
+  # Print all non-class fields.
+  print(x[attr(x, "names")])
```
Comment (brookslogan) [inline R/epidatacall.R:206]: issue?: I think this probably does both printing via messages and via regular prints to stdout.  This causes headaches when you want to build .Rmds and/or `capture.output`.  See cmu-delphi/epipredict#277.

File: R/epidatacall.R [hunk 2]
```diff
@@ -198,6 +198,14 @@ fetch_args_list <- function(
   )
 }
 
+#' @export
+print.fetch_args <- function(x, ...) {
+  stopifnot(inherits(x, "fetch_args"))
```
Comment (brookslogan) [inline R/epidatacall.R:203]: musing: I think the conventional S3 method pattern is omit this type of check, and assume that you did / could have gotten here from S3 dispatch from the generic.  Same thing probably applies to other `print` implementations added below.

File: R/model.R [hunk 3]
```diff
@@ -61,6 +61,31 @@ epirange <- function(from, to) {
...
+  stopifnot(inherits(x, "EpiRange"))
+
+  if (nchar(x$from) == 8) {
+    date_type <- "Days" # nolint: object_usage_linter
+    x$from <- as.Date(as.character(x$from), "%Y%m%d")
+    x$to <- as.Date(as.character(x$to), "%Y%m%d")
+  } else if (nchar(x$from) == 6) {
+    date_type <- "Epiweeks" # nolint: object_usage_linter
+    x$from <- format(
+      as.Date(as.character(x$from), "%Y%U"),
```
Comment (brookslogan) [inline R/model.R:75]: issue: `%U` != US epiweek convention.  Both use Sunday as start of the week, but in epiweek convention, weeks are numbered 1 to 53 and assigned to a year based on what year their Wednesday falls on.

issue: this always outputs week 0 for some reason.  Maybe `as.Date` failing due to a wday not being provided?
```
epirange(200101, 200230)

── <EpiRange> object: ────────────────────────────────────────────────────────────────────────
Epiweeks from 2001w00 to 2002w00
```

**Review by brookslogan — CHANGES_REQUESTED** [review]
Nice rounding out of the print functionality.  There appears to be some bugs with epiweek printing, plus some lower-priority annoying stuff that likely happens if you try to use this in an Rmd/qmd.

File: R/model.R (hunk 3 above)
Comment (nmdefries) [reply R/model.R]: Good catch! Avoid date conversion for simplicity.

File: R/epidatacall.R (hunk 1 above)
Comment (nmdefries) [reply R/epidatacall.R:204]: Used `cli::cli_dl` to print name-value pairs. Most `cli` functions drop item names.

**Review by dshemetov — APPROVED** [review]

File: R/model.R (hunk 3 above)
Comment (brookslogan) [reply R/model.R]: Sounds good; that's also more general.  Will be good if we add non-US-epiweeks, though the current wording ("Epiweeks") will need an update if we do.  (I don't think we have non-US-epiweeks anywhere yet (besides the `collection_weeks` which are actually specified as dates).)

**Review by brookslogan — APPROVED** [review]

**Pull request merged**